tests/InteractsWithMessengerTest.php

```diff
@@ -43,55 +44,121 @@ final class InteractsWithMessengerTest extends WebTestCase
 {
     use InteractsWithMessenger;
 
+    /**
+     * @test
+     * @group legacy
+     */
+    public function messenger_is_deprecated(): void
+    {
+        self::bootKernel();
+
+        $transport = $this->messenger();
+        $this->assertInstanceOf(TestTransport::class, $transport);
+    }
+
     /**
      * @test
      */
     public function ignore_receiver_detached_from_transport(): void
     {
         self::bootKernel(['environment' => 'with_receiver_detached']);
 
-        $transport = $this->messenger();
+        $transport = $this->transport();
         $this->assertInstanceOf(TestTransport::class, $transport);
     }
 
+    /**
+     * @test
+     */
+    public function can_interact_with_buses()
+    {
+        self::bootKernel();
+
+        self::getContainer()->get(MessageBusInterface::class)->dispatch(new MessageA());
+        $this->bus('messenger.bus.default.test-bus')->dispatched()
+            ->assertContains(MessageA::class, 1);
+    }
+
+    /**
+     * @test
+     */
+    public function interacts_with_specified_bus(): void
+    {
+        self::bootKernel(['environment' => 'multi_bus_no_routing']);
+
+        $this->bus('bus_a.test-bus')->dispatched()->assertEmpty();
+        $this->bus('bus_b.test-bus')->dispatched()->assertEmpty();
+        $this->bus('bus_c.test-bus')->dispatched()->assertEmpty();
+
+        self::getContainer()->get('bus_a.test-bus')->dispatch(new MessageA());
+        self::getContainer()->get('bus_b.test-bus')->dispatch(new MessageB());
+        self::getContainer()->get('bus_c.test-bus')->dispatch(new MessageC());
+
+        $this->bus('bus_a.test-bus')->dispatched()->assertCount(1);
+        $this->bus('bus_b.test-bus')->dispatched()->assertCount(1);
+        $this->bus('bus_c.test-bus')->dispatched()->assertCount(1);
+        $this->bus('bus_a.test-bus')->dispatched()->assertContains(MessageA::class, 1);
+        $this->bus('bus_b.test-bus')->dispatched()->assertContains(MessageB::class, 1);
+        $this->bus('bus_c.test-bus')->dispatched()->assertContains(MessageC::class, 1);
+    }
+
+    /**
+     * @test
+     */
+    public function retried_messages_are_handled(): void
+    {
+        self::bootKernel(['environment' => 'multi_bus_routing_with_fail']);
+
+        self::getContainer()->get(MessageBusInterface::class)->dispatch(new MessageA(true));
```

Comment:
no we're using php8, we can use named parameter when passing boolean like this:
```suggestion
        self::getContainer()->get(MessageBusInterface::class)->dispatch(new MessageA(fail: true));
```